app/src/main/java/com/example/gamingex/framework/impl/SingleTouchHandler.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.gamingex.framework.impl
+
+import android.view.MotionEvent
+import android.view.View
+import com.example.gamingex.framework.Input.Companion.TouchEvent
+import com.example.gamingex.framework.TouchHandler
+
+
+class SingleTouchHandler(view: View, val scaleX: Float, val scaleY: Float): TouchHandler {
+    val factory = PoolTouchFactory()
+    val touchEventPool = Pool(factory, 100)
+    var isTouched = false
+    var touchX: Int = 0
+    var touchY: Int = 0
+    val touchEvents = ArrayList<TouchEvent>()
+    val touchEventsBuffer = ArrayList<TouchEvent>()
+
+    init {
+        view.setOnTouchListener(this)
+    }
+
+    override fun onTouch(v: View?, event: MotionEvent?): Boolean {
+        synchronized(this) {
+            val touchEvent = touchEventPool.newObject()
+            when (event?.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    touchEvent.type = TouchEvent.TOUCH_DOWN
+                    isTouched = true
+                }
+                MotionEvent.ACTION_MOVE -> {
+                    touchEvent.type = TouchEvent.TOUCH_DRAGGED
+                    isTouched = true
+                }
+                MotionEvent.ACTION_CANCEL, MotionEvent.ACTION_UP -> {
+                    touchEvent.type = TouchEvent.TOUCH_UP
+                    isTouched = false
+                }
+            }
+            touchEvent.x = (event!!.x * scaleX).toInt()
+            touchX = touchEvent.x
+            touchEvent.y = (event.y * scaleY).toInt()
+            touchY = touchEvent.y
+            touchEventsBuffer.add(touchEvent)
+
+            return true
+        }
+    }
+
+    override fun isTouchDown(pointer: Int): Boolean {
+        synchronized(this) {
+            return if (pointer == 0) isTouched else false
+        }
+    }
+
+    override fun getX(pointer: Int): Int {
+        synchronized(this) {
+            return touchX
+        }
+    }
+
+    override fun getY(pointer: Int): Int {
+        synchronized(this) {
+            return touchY
+        }
+    }
+
+    override fun getTouchEvents(): List<TouchEvent> {
+        synchronized(this) {
+            for (i in touchEvents) touchEventPool.free(i)
+            touchEvents.clear()
+            touchEvents.addAll(touchEventsBuffer)
+            touchEventsBuffer.clear()
+
+            return touchEvents
```

Comment:
What's the point of `touchEventsBuffer`?
Why not just use `touchEvents` and then replace it with a new ArrayList?

Reply:
I thought it would be easier to handle the Pool interactions with a bufferArray

Reply:
It was easier for me to handle Pool interactions with a buffer array. And I thought it would be better to have val instead of var, because I wanted to reduce the GC frequency, and make the "stop the world event less possible"

Reply:
Well first of all "stop the world" is not the case for years (the JVM GC long since uses another algorithm).
Second of all, "reduce the GC frequency" - you still have to create a new `buffer` array every time because you hand it over to the user and therefore, cannot reuse it yourself so no difference there either.

Reply:
Thanks, I didn`t know that, I thought the GC is a real problem. The resources I used are a bit stale, quite an old book for Android 1.5, so I had to significantly rework it. I used it because the documentation doesn`t contain any information about game development neither in Kotlin nor in Java. 

Thanks for the comment. Next time I will think twice before I use some kind of buffers.